File: Java/MaximumTotalSubArrayValue_I.java
class MTSAV_I {
    public long maxTotalValue(int[] nums, int k) {
        int mn = nums[0];
        int mx = nums[0];

        for (int x : nums) {
            mn = Math.min(mn, x);
            mx = Math.max(mx, x);
        }

        return 1L * k * (mx - mn);
    }
}

public class MaximumTotalSubArrayValue_I {
    public static void main(String[] args) {
        
    }
}

File: C/MaximumTotalSubArrayValue_I.c
long long maxTotalValue(int* nums, int numsSize, int k) {
    int mn = nums[0];
    int mx = nums[0];

    for (int i = 1; i < numsSize; i++) {
        if (nums[i] < mn) mn = nums[i];
        if (nums[i] > mx) mx = nums[i];
    }

    return 1LL * k * (mx - mn);
}